File: smallestDifference.js
const smallestDifference = (arrayOne, arrayTwo) => {
	arrayOne.sort((a, b) => a - b)
	arrayTwo.sort((a, b) => a - b)
	let one = 0
	let two = 0
	let smallest = Infinity
	let current = Infinity
	let smallestPair = []
	while (one < arrayOne.length && two < arrayTwo.length) {
		let firstNum = arrayOne[one]
		let secondNum = arrayTwo[two]
		if (firstNum < secondNum) {
			current = secondNum - firstNum
			one++
		} else if (secondNum < firstNum) {
			current = firstNum - secondNum
			two++
		} else {
			return [firstNum, secondNum]
		}
		if (smallest > current) {
			smallest = current
			smallestPair = [firstNum, secondNum]
		}
	}
	return smallestPair
}

File: Arrays/smallest_difference.py
def smallest_difference(array1, array2):
    array1.sort()
    array2.sort()
    one = 0
    two = 0
    smallest = float('inf')
    current = float('inf')
    smallest_pair = []
    while one < len(array1) and two < len(array2):
        first = array1[one]
        second = array2[two]
        if first < second:
            current = second - first
            one += 1
        elif first > second:
            current = first - second
            two += 1
        else:
            return [first, second]
        if smallest > current:
            smallest = current
            smallest_pair = [first, second]
    return smallest_pair
